`./mach clippy -r -p webrender_traits` fails to compile

```
$ ./mach clippy -r -p webrender_traits
    Checking webrender_traits v0.0.1 (/home/oriol/src/servo/components/shared/webrender)
error[E0599]: no associated item named `Default` found for struct `surfman::NativeContext` in the current scope
   --> components/shared/webrender/rendering_context.rs:106:32
    |
106 |                 NativeContext::Default(LinuxNativeContext::Default(native_context)) => {
    |                                ^^^^^^^ associated item not found in `NativeContext`

error[E0599]: no associated item named `Default` found for struct `surfman::NativeContext` in the current scope
   --> components/shared/webrender/rendering_context.rs:109:32
    |
109 |                 NativeContext::Default(LinuxNativeContext::Alternate(native_context)) => {
    |                                ^^^^^^^ associated item not found in `NativeContext`

error[E0599]: no associated item named `Alternate` found for struct `surfman::NativeContext` in the current scope
   --> components/shared/webrender/rendering_context.rs:112:32
    |
112 |                 NativeContext::Alternate(_) => GlContext::Unknown,
    |                                ^^^^^^^^^ associated item not found in `NativeContext`

error[E0599]: no associated item named `Default` found for struct `surfman::NativeConnection` in the current scope
   --> components/shared/webrender/rendering_context.rs:136:44
    |
136 |                 surfman::NativeConnection::Default(LinuxNativeConnection::Default(connection)) => {
    |                                            ^^^^^^^ associated item not found in `NativeConnection`

error[E0599]: no associated item named `Default` found for struct `surfman::NativeConnection` in the current scope
   --> components/shared/webrender/rendering_context.rs:139:44
    |
139 |                 surfman::NativeConnection::Default(LinuxNativeConnection::Alternate(
    |                 ---------------------------^^^^^^^ associated item not found in `NativeConnection`

error[E0599]: no associated item named `Alternate` found for struct `surfman::NativeConnection` in the current scope
   --> components/shared/webrender/rendering_context.rs:142:44
    |
142 |                 surfman::NativeConnection::Alternate(_) => NativeDisplay::Unknown,
    |                                            ^^^^^^^^^ associated item not found in `NativeConnection`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `webrender_traits` (lib) due to 6 previous errors
```

Same for `./mach clippy -r -p script`, etc.

--- a/components/shared/webrender/Cargo.toml
+++ b/components/shared/webrender/Cargo.toml
@@ -26,4 +26,4 @@ webrender_api = { workspace = true }
 serde = { workspace = true }
 servo_geometry = { path = "../../geometry" }
 servo-media = { workspace = true }
-surfman = { workspace = true }
+surfman = { workspace = true, features = ["sm-x11"] }

